tools/test_utils/fullservice.py

```diff
@@ -0,0 +1,215 @@
+# TODO: This should actually be more generic so that the python CLI 
+#   can also use it as a library (or maybe tests will use the CLI's library)
+from . import constants
+import subprocess
+from urllib.parse import urlparse
+import json
+import http.client
+from typing import Tuple
+import time
+import os
+import shutil
+import pathlib
+
+class FullService:
+    def __init__(self, remove_wallet_and_ledger=False):
+        self.full_service_process = None
+        self.account_map = None
+        self.account_ids = None
+        self.request_count = 0
+        self.remove_wallet_and_ledger = remove_wallet_and_ledger
+        
+    def __enter__(self):
+        self.remove_wallet_and_ledger = True
+        self.start()
+        return self
+    
+    def __exit__(self,exc_type, exc_val, exc_tb):
+        self.stop()
+        if self.remove_wallet_and_ledger:
+            try:
+                print(f"Removing ledger/wallet dbs")
+                tmpdir = pathlib.Path('/tmp')
+                shutil.rmtree(tmpdir/'wallet-db')
+                shutil.rmtree(tmpdir/'ledger-db')
+            except Exception as e:
+                print(e) 
+        
+
+
+    def start(self):
+        cmd = ' '.join([
+            'mkdir -p /tmp/wallet-db/',
```

Comment:
This is hacky. Python has a way of creating a directory, no need to do it using a shell command.